src/IndexingMsg.chpl

```diff
@@ -312,6 +319,21 @@ module IndexingMsg
             when (DType.Bool) {
                 return sliceHelper(bool);
             }
+            when (DType.BigInt) {
+                var e = toSymEntry(gEnt,bigint);
+                var tmp = makeDistArray(slice.size, bigint);
+                ref ea = e.a;
+                // TODO figure out a way to copy more efficiently attempting
+                // to use an aggregator runs into issues with unordered copy
+                forall (elt,j) in zip(tmp, slice) {
+                    elt = ea[j];
+                }
```

Comment:
Based on @ronawho mentioning that array slicing should be pretty efficient at this point, could we just do something like `tmp = e.a[slice]`? Or is there something with the type BigInt that makes this not efficient?

Reply:
This isn't just straight slice assignment like`tmp = slice`, but random indexing into `ea` based on values in slice. I think aggregators would be appropriate here. @pierce314159 performance without aggregation is going to be quite poor and I'm really not sure if bigint can support aggregation (or really being bit-copied). Can you open an issue for me to look into this after the new year?

Reply:
Makes sense. That is why I wanted to check. My memory is a bit crowded these days.

Reply:
> This isn't just straight slice assignment like`tmp = slice`, but random indexing into `ea` based on values in slice. I think aggregators would be appropriate here. @pierce314159 performance without aggregation is going to be quite poor and I'm really not sure if bigint can support aggregation (or really being bit-copied). Can you open an issue for me to look into this after the new year?

for sure! I'll open the issue now, thanks!